pilot/pkg/config/kube/ior/route.go

```diff
@@ -555,36 +290,41 @@ func (r *route) findService(gateway *networking.Gateway) (string, string, error)
 
 	for _, ns := range namespaces {
 		// Get the list of pods that match the gateway selector
-		podList, err := r.kubeClient.CoreV1().Pods(ns).List(context.TODO(), metav1.ListOptions{LabelSelector: gwSelector.String()})
+		pods, err := r.kubeClient.CoreV1().Pods(ns).List(context.TODO(), metav1.ListOptions{LabelSelector: gwSelector.String()})
```

Comment:
This performs an actual GET on the API server instead of reading from the cache.

Reply:
Updated to use `Lister`.